filemin/setfacl.cgi

```diff
@@ -0,0 +1,55 @@
+#!/usr/local/bin/perl
+
+require './filemin-lib.pl';
+&ReadParse();
+get_paths();
+
+# Files to work on (arr)
+my @files = split(/\0/, $in{'name'});
+# Action
+my $action = $in{'action'};
+# Permission
+my $perms = $in{'perms'};
+# User
+my $user = $in{'user'};
+# Group
+my $group = $in{'group'};
+# Recursive
+my $recursive = $in{'recursive'} ? " -R" : "";
+# Manual
+my $extra = $in{'manual'};
+# Apply to (arr)
+my @apply_to = split(/\0/, $in{'apply_to'});
+my @types;
+foreach my $type (@apply_to) {
+    if ($user && $type eq 'u') {
+        push(@types, "u:${user}:${perms}");
+        }
+    if ($group && $type eq 'g') {
+        push(@types, "g:${group}:${perms}");
+        }
+    if ($type =~ /^m|o$/) {
+        push(@types, "${type}::${perms}");
+        }
+    }
+my $cmd = &has_command('setfacl');
+error($text{'acls_error'}) if (!$cmd);
+my $types;
+$types = join(',',@types) if (@types);
+$types .= " $extra" if ($extra);
+my $args = "$action $types $recursive";
+$args =~ s/\s+/ /g;
+$args = &trim($args);
+$args =~ s/[\`\$\;\/\'\"\?\%\&\#\*\(\)\+]//g;
```

Comment:
Why not use quotemeta instead of this?

Reply:
Well, it's just doesn't work with `quotemeta`. The command just fails.

Reply:
Ok but what's the purpose of this regular expression?

Reply:
To make sure that line injection won't be possible.

Reply:
So why not use the standard Perl `quotemeta` function? It will do all the escaping needed in this situation ...

Reply:
> It will do all the escaping needed in this situation

Yeah, fixed [here](https://github.com/webmin/webmin/pull/1846/commits/31eb2180944af65acda9313e830d03444e1e01ae). However, you need to avoid running `quotemeta` on spaces in the params' line, otherwise (for some reason) it fails.

Reply:
You have to call quotemeta on each param separately